server/src/server.ts

```diff
@@ -13,6 +13,8 @@ import { BashCompletionItem, CompletionItemDataType } from './types'
 import { uniqueBasedOnHash } from './util/array'
 import { getShellDocumentation } from './util/sh'
 
+const PARAMETER_EXPANSION_PREFIXES = new Set(['$', '${', '${#'])
+
```

Comment:
this variable is used to test if:

```js
    const shouldCompleteOnVariables = word
      ? PARAMETER_EXPANSION_PREFIXES.has(word)
      : false
```

This check sets `shouldCompleteOnVariables` to true only for the first two, and only if the user hasnt typed anything else. I've done some tests, and I can't find a case where `${#` gets detected.